flood_adapt/object_model/tipping_point.py

```diff
@@ -0,0 +1,328 @@
+        # TODO: check this
+
+        check = all(results_toml.exists() for result in [results_toml, results_csv])
+        if check:
+            with open(results_toml, "rb") as fp:
+                self.results = tomli.load(fp)
+
+    def check_scenarios(self):
+        """Check which scenarios are needed for this tipping point calculation
+        and if they have already been created"""
+        # calculating scenarios per slr level
+        scenarios_calc = {
+            f"slr_{slr}": {
+                "name": f"slr_{slr}",
+                "event": self.attrs.event_set,
+                "projection": self.attrs.projection,
+                "strategy": self.attrs.strategy,
+                "sealevelrise": slr,
+                "tipping_point_metric": self.attrs.tipping_point_metric,
+            }
+            for slr in self.attrs.sealevelrise
+        }
+
+        scenarios_avail = []
+        for scenario_path in list(
+            self.database_input_path.joinpath("scenarios").glob("*")
+        ):
+            scenarios_avail.append(
+                Scenario.load_file(scenario_path.joinpath(f"{scenario_path.name}.toml"))
+            )
+
+        # Check if any of the needed scenarios are already there
+        for scenario in scenarios_calc.keys():
+            scn_dict = scenarios_calc[scenario].copy()
+            scn_dict["name"] = scenario
+            scenario_obj = Scenario.load_dict(scn_dict, self.database_input_path)
+            created = [
+                scn_avl for scn_avl in scenarios_avail if scenario_obj == scn_avl
+            ]
+            if len(created) > 0:
+                scenarios_calc[scenario]["scenario created"] = created[0].attrs.name
+                if created[0].init_object_model().direct_impacts.has_run:
+                    scenarios_calc[scenario]["scenario run"] = True
+                else:
+                    scenarios_calc[scenario]["scenario run"] = False
+            else:
+                scenarios_calc[scenario]["scenario created"] = "No"
+                scenarios_calc[scenario]["scenario run"] = False
+
+        df = pd.DataFrame(scenarios_calc).T
+        self.scenarios = df.astype(
+            dtype={
+                "event": "str",
+                "projection": "str",
+                "strategy": "str",
+                "scenario created": "str",
+                "scenario run": bool,
+            }
+        )
+        return self.scenarios
+
+    # standard functions
+    def load_file(
+        filepath: Union[str, Path], database_input_path: Union[str, os.PathLike]
+    ) -> "TippingPoint":
+        """create risk event from toml file"""
+
+        obj = TippingPoint(database_input_path)
+        with open(filepath, mode="rb") as fp:
+            toml = tomli.load(fp)
+        obj.attrs = TipPointModel.model_validate(toml)
+        return obj
+
+    def load_dict(
+        dct: Union[str, Path], database_input_path: Union[str, os.PathLike]
```

Comment:
Every load_dict function in the code base has one argument which is the dict you are trying to load. So we definitely want to keep consistent with that. (in addition to my earlier comments)